### ISSUE TITLE
Log spam of "Received a ServerHelloDone handshake message while expecting [CertificateRequest]" in 0.20.4

### ISSUE THREAD

**Issue by jbg** [rustls/rustls#1012]
[This log message](https://github.com/rustls/rustls/blob/main/rustls/src/check.rs#L66) is triggering for every HTTPS request we make to a certain upstream API, as of 0.20.4.

The message, logged at WARN level, is:

> Received a ServerHelloDone handshake message while expecting [CertificateRequest]

It seems erroneous given that it's valid to receive no `CertificateRequest` before `ServerHelloDone`.

**Comment by ctz** [comment]
Thanks for the report, this is indeed spurious logging and a regression in 0.20.3 and 0.20.4.

**Comment by TjeuKayim** [comment]
The URL https://httpbin.org/ is affected by this bug, while the URL https://example.com/ is not.

Minimal code to reproduce this with the reqwest crate:

```rs
reqwest::get("https://httpbin.org/").await?;
```

Logs the following warning:

```log
[WARN  rustls::check] Received a ServerHelloDone handshake message while expecting [CertificateRequest]
```

**Comment by djc** [comment]
Should be fixed in #1013.
